yarn-project/noir-libs/value-note/src/utils.nr

```diff
@@ -1,184 +1,125 @@
+use dep::std::option::Option;
 use dep::aztec::context::PrivateContext;
 // docs:start:encrypted_import
 
 use dep::aztec::log::emit_encrypted_log;
 
 // docs:end:encrypted_import
-use dep::aztec::note::note_getter_options::NoteGetterOptions;
+use dep::aztec::note::note_getter_options::{NoteGetterOptions, SortOrder};
 use dep::aztec::oracle::get_public_key::get_public_key;
 use dep::aztec::state_vars::set::Set;
-use dep::aztec::types::point::Point;
 use crate::{
     filter::filter_notes_min_sum,
     value_note::{ValueNote, VALUE_NOTE_LEN},
 };
 
-fn spend_notes(
+// Sort the note values (0th field) in descending order.
+// Pick the fewest notes whose sum is equal to or greater than `amount`.
+fn create_note_getter_options_for_decreasing_balance(amount: Field) -> NoteGetterOptions<ValueNote, VALUE_NOTE_LEN, Field> {
+    NoteGetterOptions::with_filter(filter_notes_min_sum, amount).sort(0, SortOrder.DESC)
+}
+
+// Creates a new note for the recipient.
+// Inserts it to the recipient's balance.
+fn increase_balance(
```

Comment:
A slight nitpick with the name `increase_balance`: the word `balance` isn't related to a `ValueNote` - the word `balance` is a state variable naming choice of the apps which import the ValueNote. The ValueNote could be used to represent something else, where the word "balance" doesn't make sense.

Maybe `increase_value` and `decrease_value`? Maybe `increment` and `decrement`?

Reply:
But it does take a "balance" as parameter, and it's not increasing a note's value? Naming is so hard 🥲 
Will change them to `increment` and `decrement`. I like that they are more neutral!